TimJohnson/Homework_Week1.js

```diff
@@ -0,0 +1,139 @@
+1o)	"0"+1 == 1;	true;	the left evaluates to '01' which is converted to a number which makes the zero a placeholder and the value = 1 same as the right side
+1p)	(typeof (x+1))===(typeof x);	true (if x is a number) false if x is string or undefined;  both sides evaluate to a 'number'
+1q)	(x*1 == x) || ((typeof x) != "number");	false;	if either of the conditional ORs evaluate to something other than a number the equation is true(.  
+	(x*1 == x) equals boolean true which is not equal to 'number'
+1p)	(x=(typeof (x+(typeof x))))==x;	true;	evaluating the inner typeofs, we get x = stringwhich is == x
+
+2a)
+	var x; 
+	var y; 
+	var z; 
+	var mean;
+	mean = (x + y + z) / 3;  // Calculates the average (mean) value
+	countlog(mean);
+
+2b)
+	var x; 
+	var y; 
+	var z; 
+	var mean;
+	mean = (x + y + z) / 3;  // Calculate mean of three variables
+	x = (mean - x) / 2 + x; // Calculates new value of x to be halfway to mean
+	y = (mean - y) / 2 + y; // Calculates new value of y to be halfway to mean
+	z = (mean - z) / 2 + z; // Calculates new value of z to be halfway to mean
+	
+3a)
+	var l; 
+	var r;
+	var t; 
+	var b;
+	var area;
+	area = (r - l) * (t - b); // Calculates the are of rectangle (l, r, t, b)
+	
+3b)
+	var l; 
+	var r;
+	var t; 
+	var b;
+	var shape;
+	if ((r - l) < (t-b)) // Tests if width is less than height of rectangle
+		shape = 'true'; // Width is less than height
+	else 
+		shape = 'false'; // Width is greater than height
+
+3c)
+	var l; 
+	var r;
+	var t; 
+	var b;
+	var circumference;
+	var pi = 3.1415
+	if ((r - l) > (t - b)) // Tests which has the greater value, width or height
+		circumference = 2 * pi * (t - b)/2; // If width is greater, use height to calculate a circle to fit inside the rectangle
+	else 
+		circumference = 2 * pi * (r - l)/2; // If height is greater, use width to calculate a circle to fit inside the rectangle
+
+3d)
+	var l; 
+	var r;
+	var t; 
+	var b;
+	var area;
+	var pi = 3.1415;
+	if ((r - l) < (t - b)) // Tests which has the greater value, width or height
+		area = pi * ((t - b) / 2) * ((t - b) / 2);  // If width is greater, use width to calculate a circle size to fit around entire rectangle
+	else 
+		area = pi * ((r - l) / 2) * ((r - l) / 2); // If width is less, use height to calculate a circle size to fit around entire rectangle
+		
+3e)
+	var l; 
+	var r;
+	var t; 
+	var b;
+	var l2; 
+	var r2;
+	var t2; 
+	var b2;
+	l2 = ((r - l) / 3) + l;  // Returns the left coordinate
+	r2 = r - ((r - l) / 3);  // Returns the right coordinate
+	t2 = t - ((t - b) / 3);	 // Returns the top coordinate
+	b2 = ((t - b) / 3) + b;  // Returns the bottom coordinate
+	
+4a)
+	var R;
+	var C;
+	var N;
+	N = (R * 8) + C; // Calculate Square num when Row num and Col num are known
+	
+4b)
+	var R;
+	var C;
+	var N;
+	R = (N - C) / 8; // Calculate Row num when Col num and Square num are known
```

Comment:
For this one, you would want to use math.floor, which is a function that rounds the solution down to the nearest whole number. Here is Dan's answer: r = Math.floor(n/8); So, essentially, you would divide the selected number on the checkerboard by 8, and then round down, which would give you the correct row.